core/src/main/scala/zio/config/ConfigDocs.scala

```diff
@@ -1,16 +1,21 @@
 package zio.config
 
+import zio.config
+
 sealed trait ConfigDocs[+K, +V]
 object ConfigDocs {
   final case class Leaf[V](sources: Sources, descriptions: List[String], value: Option[V] = None)
       extends ConfigDocs[Nothing, V]
 
   final case class Sources(set: Set[String])
 
-  final case object Empty                                                               extends ConfigDocs[Nothing, Nothing]
-  final case class NestedPath[K, V](path: K, docs: ConfigDocs[K, V])                    extends ConfigDocs[K, V]
-  final case class Both[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])          extends ConfigDocs[K, V]
-  final case class OneOf[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])         extends ConfigDocs[K, V]
-  final case class Sequence[K, V](element: List[ConfigDocs[K, V]])                      extends ConfigDocs[K, V]
-  final case class DynamicMap[K, V](element: scala.collection.Map[K, ConfigDocs[K, V]]) extends ConfigDocs[K, V]
+  final case object Empty                                                       extends ConfigDocs[Nothing, Nothing]
+  final case class NestedPath[K, V](path: K, docs: ConfigDocs[K, V])            extends ConfigDocs[K, V]
+  final case class Both[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])  extends ConfigDocs[K, V]
+  final case class OneOf[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V]) extends ConfigDocs[K, V]
+  final case class Sequence[K, V](element: List[ConfigDocs[K, V]])              extends ConfigDocs[K, V]
+  final case class DynamicMapInit[K, V](element: config.ConfigDocs[K, V])       extends ConfigDocs[K, V]
+  // During reporting, we know the dynamic keys; Fixme; Make it better orthogonal
+  final case class DynamicMap[K, V](element: Map[K, config.ConfigDocs[K, V]]) extends ConfigDocs[K, V]
```

Comment:
This is an immediate fix, however, the we still have an issue https://github.com/zio/zio-config/issues/287.

We use `generateDocs` as the first step to reporting, where values gets injected into the structure used for generating docs. Meaning `def report(desc, value) = generateDocs(description).map(injectValueIntoTree(value))`

With the introduction of map, the keys become dynamic and is only available only when reporting.